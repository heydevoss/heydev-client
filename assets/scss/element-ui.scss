// Import Custom Variables
@import "variables";

/* icon font path, required */
$--font-path: '~element-ui/lib/theme-chalk/fonts';

@import "~element-ui/packages/theme-chalk/src/index";