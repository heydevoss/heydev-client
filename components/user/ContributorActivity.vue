<template>
  <div class="block">
    <br /><el-timeline>
      <el-timeline-item
        v-for="(activity, index) in activities"
        :key="index"
        :icon="activity.icon"
        :type="activity.type"
        :color="activity.color"
        :size="activity.size"
        :timestamp="activity.timestamp"
      >
        {{ activity.content }}
      </el-timeline-item>
    </el-timeline>
  </div>
</template>

<script>
export default {
  name: 'ContributorActivity',
  data() {
    return {
      activities: [
        {
          content: 'New Pull Request on Tamburetei',
          timestamp: '2019-06-12 20:46',
          size: 'large',
          type: 'primary',
          color: 'grey',
          icon: 'el-icon-s-promotion'
        },
        {
          content: 'Commented on IssueAi #48 issue',
          timestamp: '2019-06-12 16:05',
          color: 'grey',
          icon: 'el-icon-more',
          size: 'large'
        },
        {
          content: 'Star in Roadmap',
          timestamp: '2019-06-11 13:56',
          size: 'large',
          icon: 'el-icon-star-on',
          color: 'grey'
        },
        {
          content: 'Commit on IssueAi',
          timestamp: '2019-06-10 19:36',
          icon: 'el-icon-circle-plus-outline',
          color: 'grey',
          size: 'large'
        }
      ]
    }
  }
}
</script>
