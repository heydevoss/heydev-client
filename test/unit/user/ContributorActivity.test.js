import Vue from 'vue'
import { shallowMount } from '@vue/test-utils'
import ContributorActivity from '@/components/user/ContributorActivity'
import { Timeline, TimelineItem } from 'element-ui'

Vue.use(Timeline)

describe('<ContributorActivity/> Component', () => {
  const wrapper = shallowMount(ContributorActivity, {
    stubs: {
      'el-timeline': Timeline,
      'el-timeline-item': TimelineItem
    }
  })

  it('should be a Vue instance and render correctly', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('should contain timeline', () => {
    expect(wrapper.contains(Timeline)).toBe(true)
  })

  it('should contain an array of activities', () => {
    expect(wrapper.vm.activities)
  })
})
