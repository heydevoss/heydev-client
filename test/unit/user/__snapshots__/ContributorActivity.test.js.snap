// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContributorActivity/> Component should be a Vue instance and render correctly 1`] = `
<div
  class="block"
>
  <br />
  <ul
    class="el-timeline"
  >
    <li
      class="el-timeline-item"
    >
      <div
        class="el-timeline-item__tail"
      />
      <div
        class="el-timeline-item__node el-timeline-item__node--large el-timeline-item__node--primary"
        style="background-color: grey;"
      >
        <i
          class="el-timeline-item__icon el-icon-s-promotion"
        />
      </div>
      <!---->
      <div
        class="el-timeline-item__wrapper"
      >
        <!---->
        <div
          class="el-timeline-item__content"
        >
          
      New Pull Request on Tamburetei
    
        </div>
        <div
          class="el-timeline-item__timestamp is-bottom"
        >
          
      2019-06-12 20:46
    
        </div>
      </div>
    </li>
    <li
      class="el-timeline-item"
    >
      <div
        class="el-timeline-item__tail"
      />
      <div
        class="el-timeline-item__node el-timeline-item__node--large el-timeline-item__node--"
        style="background-color: grey;"
      >
        <i
          class="el-timeline-item__icon el-icon-more"
        />
      </div>
      <!---->
      <div
        class="el-timeline-item__wrapper"
      >
        <!---->
        <div
          class="el-timeline-item__content"
        >
          
      Commented on IssueAi #48 issue
    
        </div>
        <div
          class="el-timeline-item__timestamp is-bottom"
        >
          
      2019-06-12 16:05
    
        </div>
      </div>
    </li>
    <li
      class="el-timeline-item"
    >
      <div
        class="el-timeline-item__tail"
      />
      <div
        class="el-timeline-item__node el-timeline-item__node--large el-timeline-item__node--"
        style="background-color: grey;"
      >
        <i
          class="el-timeline-item__icon el-icon-star-on"
        />
      </div>
      <!---->
      <div
        class="el-timeline-item__wrapper"
      >
        <!---->
        <div
          class="el-timeline-item__content"
        >
          
      Star in Roadmap
    
        </div>
        <div
          class="el-timeline-item__timestamp is-bottom"
        >
          
      2019-06-11 13:56
    
        </div>
      </div>
    </li>
    <li
      class="el-timeline-item"
    >
      <div
        class="el-timeline-item__tail"
      />
      <div
        class="el-timeline-item__node el-timeline-item__node--large el-timeline-item__node--"
        style="background-color: grey;"
      >
        <i
          class="el-timeline-item__icon el-icon-circle-plus-outline"
        />
      </div>
      <!---->
      <div
        class="el-timeline-item__wrapper"
      >
        <!---->
        <div
          class="el-timeline-item__content"
        >
          
      Commit on IssueAi
    
        </div>
        <div
          class="el-timeline-item__timestamp is-bottom"
        >
          
      2019-06-10 19:36
    
        </div>
      </div>
    </li>
  </ul>
</div>
`;
